### DISCUSSION TITLE
Mengubah tipe data

### DISCUSSION THREAD

**Discussion by yonvisatria20** [teamalgoritma/community#263] [Q&A] [answered]
Halo saya ingin bertanya , Saya punya kolom dengan tipe data numerical yaitu 1-12 yang mempresentasikan bulan, setelah saya coba ubah ke karakter ( Januari-Desember) itu bisa, namun saya ingin membuat agar tetap terbaca sebagai numeric. Contoh : 1 yaitu bulan Januari, ekspeknya agar terbaca nya sebagai index 1 bukan 10 (urutan huruf J di alfabet). Fungsi apa yg dipakai untuk membuat seperti itu? Terimakasih sebelumnya teman2 mentor.

**Comment by VictorNugraha** [comment, marked as answer]
Hai, @yonvisatria20 !

Jika kita ingin mengubah value yang bersifat karakter menjadi tipe data numeric, setau saya masih belum bisa.

Untuk dapat tetap mengurutkan sebuah nilai karakter yang berulang (factor) tanpa berdasarkan urutan alfabetnya, bisa mencoba fungsi  **_factor()_** untuk mengurutkan levels factor dalam sebuah kolom. Pada fungsi tersebut kita bisa isi dengan 2 parameter, yaitu:

- _x_ -> nama kolom yang ingin diatur levels factornya.
- _levels_ -> urutan levels factor yang diharapkan.

Berikut contoh codenya:

`df$nama_kolom<- factor(x = df$nama_kolom, levels = c("urutan_1", "urutan_2", "urutan_3"))`

Demikian infromasi yang dapat saya sampaikan, semoga menjawab pertanyaan yang diberikan. Selamat mencoba :D
